beta/experience-platform-extension/tutorials/tutorial-4-rules-xdm-events.md

```diff
@@ -0,0 +1,154 @@
+
+Set the `LAUNCH_ENVIRONMENT_FILE_ID` to the copied Environment File ID in the `MainApp` \(Android\) / `AppDelegate` \(iOS\) file.
+
+Set the `PRODUCT_REVIEW_DATASET_ID` to the dataset identifier for Product reviews created in Assignment 3 in the `EdgeTab` (Android) / `EdgeViewController` (iOS) file.
+
+In the same file, set the `TENANT_ID` to the org identifier as described in the Product reviews schema.
+
+## Attach data to XDM events
+
+### Update the Product reviews schema
+
+1. In the browser, navigate to [Adobe Experience Platform](https://experience.adobe.com/platform) and login with your credentials.
+
+2. Update the `Product reviews` schema created in Assignment 3 by adding two new fields.
+
+   - From the left panel, select `Schemas` and then click on the `Product reviews` schema you have created for the previous assignment.
+
+   - Click on the `Product review` mixin.
+
+   - Click on the plus icon `+` next to the schema name and add the following fields. Click `Apply` after each field added.
+
+     | Field name   | Display name  | Type   | Required |
+     | :----------- | :------------ | :----- | -------- |
+     | ratingSource | Rating Source | String | No       |
+     | ratingDate   | Rating Date   | String | No       |
+
+   - Click `Save` to update the schema.
+
+### Create a mobile rule in Adobe Experience Launch
+
+1. Navigate to [Adobe Experience Launch](https://experience.adobe.com/launch) and select the mobile property you created in the previous assignments.
+
+2. Click on `Extensions` from the left panel and update the `Adobe Experience Platform Edge` extension to latest version.
+
+3. Click on `Data elements` from the left panel and the following data elements:
+
+   1. Create `App ID` data element:
+      - Click on the `Create New Data Element` button.
+      - Select the `Mobile Core` Extension and `App ID` Data Element Type.
+      - Set the name for this data element to `App ID` and click `Save`.
+   2. Create `Operating system version` data element:
+      - Click on the `Add Data Element` button.
+      - Select the `Mobile Core` Extension and `OS Version` Data Element Type.
+      - Set the name for this data element to `Operating system version` and click `Save`.
+   3. Create `Event timestamp` data element:
+      - Click on the `Add Data Element` button.
+      - Select the `Mobile Core` Extension and `Adobe Experience Platform Timestamp` Data Element Type.
+      - Set the name for this data element to `Event timestamp` and click `Save`.
+   
+4. Click on `Rules` from the left pannel and create the following rule:
+
+   - Click `Create New Rule`.
```

Comment:
"`Add Rule`"

Reply:
Ah, ignore comment. "Create New Rule" is correct when there are no existing rules. :)